src/nvim/tui/tui.c

```diff
@@ -350,6 +351,10 @@ static void tui_resize(UI *ui, int width, int height)
   data->scroll_region.left = 0;
   data->scroll_region.right = width - 1;
   data->row = data->col = 0;
+  // try to resize the terminal window
+  char r[16];  // enough for 9999x9999
+  snprintf(r, sizeof(r), "\x1b[8;%d;%dt", height, width);
```

Comment:
Thanks @ZyX-I. I missed the word "pixels", I was focused on "window" vs "text area" and I took the latter to mean something else.